src/graphics/program-lib/programs/standard.js

```diff
@@ -231,1552 +172,148 @@ const standard = {
-                    } else {
-                        if (options.clearCoat > 0) code += "       ccSpecularLight += getLightSpecularCC() * dAtten * light" + i + "_color" + (usesCookieNow ? " * dAtten3" : "") + ";\n";
-                        if (options.useSpecular) code += "       dSpecularLight += getLightSpecular() * dAtten * light" + i + "_color" + (usesCookieNow ? " * dAtten3" : "") + ";\n";
-                    }
-                }
-
-                if (lightType !== LIGHTTYPE_DIRECTIONAL) {
-                    code += "   }\n"; // BRANCH END
-                }
-
-                code += "\n";
-            }
-
-            // clustered lighting
-            if (options.clusteredLightingEnabled && lighting) {
-                usesLinearFalloff = true;
-                usesInvSquaredFalloff = true;
-                hasPointLights = true;
-                code += '   addClusteredLights();\n';
-            }
-
-            if (hasAreaLights) {
-                // specular has to be accumulated differently if we want area lights to look correct
-                if (options.clearCoat > 0) {
-                    code += "   ccSpecularity = 1.0;\n";
-                }
-                if (options.useSpecular) {
-                    code += "   dSpecularity = vec3(1);\n";
-                }
-            }
-
-            if (reflections && options.refraction) {
-                code += "   addRefraction();\n";
-            }
-        }
-        code += "\n";
-
-        if (useAo) {
-            if (options.occludeDirect) {
-                code += "    occludeDiffuse();\n";
-            }
-            if (options.occludeSpecular) {
-                code += "    occludeSpecular();\n";
+            // all other passes require only opacity
+            if (options.alphaTest) {
+                decl += "float dAlpha;\n";
+                code += this._addMap("opacity", "opacityPS", options, litShader.chunks);
+                code += litShader.chunks.alphaTestPS;
+                func += "getOpacity();\n";
+                func += "alphaTest(dAlpha);\n";
             }
         }
 
-        if (options.opacityFadesSpecular === false) {
-            if (options.blendType === BLEND_NORMAL || options.blendType === BLEND_PREMULTIPLIED) {
-                code += "float specLum = dot((dSpecularLight + dReflection.rgb * dReflection.a) * dSpecularity, vec3( 0.2126, 0.7152, 0.0722 ));\n";
-                code += "#ifdef CLEARCOAT\n specLum += dot(ccSpecularLight * ccSpecularity + ccReflection.rgb * ccReflection.a * ccSpecularity, vec3( 0.2126, 0.7152, 0.0722 ));\n#endif\n";
-                code += "dAlpha = clamp(dAlpha + gammaCorrectInput(specLum), 0.0, 1.0);\n";
-            }
-            code += "dAlpha *= material_alphaFade;\n";
-        }
+        decl = `//-------- frontend begin\n${decl}//-------- frontend end\n`;
```

Comment:
I'd probably change these comments to 'frontend decl begin`, `frontend code begin` ... instead of all being the same

Reply:
done, thanks